common/nymcoconut/src/scheme/issuance.rs

```diff
@@ -0,0 +1,397 @@
+            &cm_bytes,
+            CoconutError::Deserialization(
+                "Failed to deserialize compressed commitment".to_string(),
+            ),
+        )?;
+        j += commitment_bytes_len;
+
+        let cm_hash_bytes = bytes[j..j + commitment_hash_bytes_len].try_into().unwrap();
+        let commitment_hash = try_deserialize_g1_projective(
+            &cm_hash_bytes,
+            CoconutError::Deserialization(
+                "Failed to deserialize compressed commitment hash".to_string(),
+            ),
+        )?;
+        j += commitment_hash_bytes_len;
+
+        let c_len = u64::from_le_bytes(bytes[j..j + 8].try_into().unwrap());
+        j += 8;
+        if bytes[j..].len() < c_len as usize * 96 {
+            return Err(CoconutError::DeserializationMinLength {
+                min: c_len as usize * 96,
+                actual: bytes[56..].len(),
+            });
+        }
+
+        let mut private_attributes_ciphertexts = Vec::with_capacity(c_len as usize);
+        for i in 0..c_len as usize {
+            let start = j + i * 96;
+            let end = start + 96;
+            private_attributes_ciphertexts.push(Ciphertext::try_from(&bytes[start..end])?)
+        }
+
+        let pi_s = ProofCmCs::from_bytes(&bytes[j + c_len as usize * 96..])?;
+
+        Ok(BlindSignRequest {
+            commitment,
+            commitment_hash,
+            private_attributes_ciphertexts,
+            pi_s,
+        })
+    }
+}
+
+impl Bytable for BlindSignRequest {
+    fn to_byte_vec(&self) -> Vec<u8> {
+        let cm_bytes = self.commitment.to_affine().to_compressed();
+        let cm_hash_bytes = self.commitment_hash.to_affine().to_compressed();
+        let c_len = self.private_attributes_ciphertexts.len() as u64;
+        let proof_bytes = self.pi_s.to_bytes();
+
+        let mut bytes = Vec::with_capacity(48 + 48 + 8 + c_len as usize * 96 + proof_bytes.len());
+
+        bytes.extend_from_slice(&cm_bytes);
+        bytes.extend_from_slice(&cm_hash_bytes);
+        bytes.extend_from_slice(&c_len.to_le_bytes());
+        for c in &self.private_attributes_ciphertexts {
+            bytes.extend_from_slice(&c.to_bytes());
+        }
+
+        bytes.extend_from_slice(&proof_bytes);
+
+        bytes
+    }
+
+    fn try_from_byte_slice(slice: &[u8]) -> Result<Self> {
+        BlindSignRequest::from_bytes(slice)
+    }
+}
+
+impl Base58 for BlindSignRequest {}
+
+impl BlindSignRequest {
+    fn verify_proof(&self, params: &Parameters, pub_key: &elgamal::PublicKey) -> bool {
+        self.pi_s.verify(
+            params,
+            pub_key,
+            &self.commitment,
+            &self.private_attributes_ciphertexts,
+        )
+    }
+
+    pub fn get_commitment_hash(&self) -> G1Projective {
+        self.commitment_hash
+    }
+
+    // TODO: perhaps also include pi_s.len()?
+    // to be determined once we implement serde to make sure its 1:1 compatible
+    // with bincode
+    // cm || c.len() || c || pi_s
```

Comment:
Another outdated comment as we seem to have added commitment hash to the blind sign request